components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -161,6 +161,19 @@ bool DeletePathOnFileTaskRunner(const base::FilePath& path) {
   return base::DeleteFile(path);
 }
 
+bool ClearHistoryOnExitOnFileTaskRunner(
+    const base::FilePath& ads_service_path,
+    const bool user_has_joined_brave_rewards) {
+  // Do not clear history for Rewards users from the database, as it is
```

Comment:
If we're not going to honor the user's request to delete Brave Ads data, then I think in that case the check box should be grayed out. Otherwise we're not doing what we "promised" to do by offering the checkbox.

Reply:
The concern is fixed by adding `Clear Brave Ads data...` link to the settings for non-Rewards users. 
When clicking this link all Brave Ads files and preferences are deleted.

If user joined Brave Rewards then `Clear Brave Ads data...` link is replaced by `Reset Brave Rewards data...`, which reset rewards and clears all Brave Ads files and preferences as well.